homeassistant/util/dt.py

```diff
@@ -92,7 +94,7 @@ def as_local(dattim: dt.datetime) -> dt.datetime:
 
 def utc_from_timestamp(timestamp: float) -> dt.datetime:
     """Return a UTC time from a timestamp."""
-    return dt.datetime.utcfromtimestamp(timestamp).replace(tzinfo=UTC)
+    return UTC.localize(dt.datetime.utcfromtimestamp(timestamp))
```

Comment:
Is this the same? Is a non timezone aware date that is being localized treated as having been UTC time? Or would it treat it as local time?

Reply:
Here is the implementation of UTC.localize
```
    def localize(self, dt, is_dst=False):
        '''Convert naive time to local time'''
        if dt.tzinfo is not None:
            raise ValueError('Not naive datetime (tzinfo is already set)')
        return dt.replace(tzinfo=self)
```